src/collective/behavior/banner/browser/viewlets.py

```diff
@@ -96,3 +102,44 @@ def random_banner(self):
 
         random.shuffle(banners)
         return banners
+
+    def getVideoEmbedMarkup(self, obj):
+        """ Build an iframe from a YouTube or Vimeo share url
+        """
+        # https://www.youtube.com/watch?v=Q6qYdJuWB6w
+        YOUTUBE_TEMPLATE = '''
+            <iframe
+                width="660"
+                height="495"
+                src="//www.youtube-nocookie.com/embed/{1}?showinfo=0"
+                frameborder="0"
+                allowfullscreen>
+            </iframe>
+        '''
+        # https://vimeo.com/75721023
+        VIMEO_TEMPLATE = '''
+            <iframe
+                src="//player.vimeo.com/video/{0}?title=0&amp;byline=0&amp;portrait=0"
+                width="660"
+                height="371"
+                frameborder="0"
+                webkitallowfullscreen
+                mozallowfullscreen
+                allowfullscreen>
+            </iframe>
+        '''
+        url = obj['banner_url']
+        from urlparse import urlparse
+        try:
+            parsed = urlparse(url)
+        except AttributeError:
+            return ''
+        path = parsed.path.replace('/', '')
+        videoId = parsed.query.replace('v=', '')
+        if 'youtube' in parsed.netloc:
+            template = YOUTUBE_TEMPLATE
+        elif 'vimeo' in parsed.netloc:
+            template = VIMEO_TEMPLATE
+        else:
+            return ''
+        return template.format(path, videoId)
```

Comment:
@fulv I don't really get that part with the {0} and {1}. Do you include the path for vimeo and the videoId for youtube? Can you add a comment that explains it please?